### ISSUE TITLE
Problem when using Record + optional types

### ISSUE THREAD

**Issue by DLMR** [esdoc/esdoc#152]
There is an error when using [Record](https://esdoc.org/tags.html#record) along with different types. Or maybe this is not something that is supported by choice?

__Example__
```javascript
/**
 * @param {{bar: (string|string[])}} options - this is object param.
 */
export function foo(options = {}) {}
```

The following errors is thrown when bulding the docs.
````
esdoc/out/src/Publisher/Builder/DocBuilder.js:469
            throw _iteratorError2;
            ^

SyntaxError: Invalid regular expression: /[~]\{bar: (string$/: Unterminated group
    at RegExp (native)
````

Looked a bit more into this and the issue above is because of the parenthesis, however removing them  results in a lint error about invalid JSDoc (might not be relevant for ESDoc). 

```javascript
/**
 * @param {{bar: string|string[]}} options - this is object param.
 */
export function foo(options = {}) {}
```

The output becomes.

Name | Type | Attribute | Description
------------ | -------------|---------|-----------
options |	{bar: string \| string[]} |		| this is object param.

And not the expected.

Name | Type | Attribute | Description
------------ | -------------|---------|-----------
options |	{bar: [string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) \| [string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)[]} |		| this is object param.

**Comment by h13i32maru** [comment]
@DLMR Hi. Thanks for this issue. This is ESDoc bug. I will fix this.

**Comment by h13i32maru** [comment]
I fixed this issue at v0.4.2.
